### PR TITLE
[RISCV][GISEL] Regbank select and instruction select for scalable vector G_ICMP

### PR DESCRIPTION
**Pull request by michaelmaitland**
Register bank selection chooses VRB when the definition is a vector type, since this is what the legalizer lets through. Otherwise this is a G_ICMP with all GPR operands.

Instruction selection works out of the box due to SelectionDAG patterns. We add tests here to demonstrate it works.

### CODE REVIEW

File: llvm/lib/Target/RISCV/GISel/RISCVRegisterBankInfo.cpp [hunk 1]
```diff
@@ -487,6 +487,19 @@ RISCVRegisterBankInfo::getInstrMapping(const MachineInstr &MI) const {
     OpdsMapping[1] = GPRValueMapping;
     break;
   }
+  case TargetOpcode::G_ICMP: {
+    if (MRI.getType(MI.getOperand(0).getReg()).isVector()) {
```
Comment (topperc) [inline llvm/lib/Target/RISCV/GISel/RISCVRegisterBankInfo.cpp:491]: Can you just add vector support to the default case?

File: llvm/test/CodeGen/RISCV/GlobalISel/instruction-select/rvv/icmp.mir [hunk 2]
```diff
@@ -0,0 +1,534 @@
...
+    ; RV32I-NEXT: $v8 = COPY [[PseudoVMSLTU_VV_MF4_]]
+    ; RV32I-NEXT: PseudoRET implicit $v8
+    ;
+    ; RV64I-LABEL: name: icmp_nxv2i8
+    ; RV64I: [[DEF:%[0-9]+]]:vr = IMPLICIT_DEF
+    ; RV64I-NEXT: [[PseudoVMSLTU_VV_MF4_:%[0-9]+]]:vr = PseudoVMSLTU_VV_MF4 [[DEF]], [[DEF]], -1, 3 /* e8 */
+    ; RV64I-NEXT: $v8 = COPY [[PseudoVMSLTU_VV_MF4_]]
+    ; RV64I-NEXT: PseudoRET implicit $v8
+    %0:vrb(<vscale x 2 x s8>) = G_IMPLICIT_DEF
+    %1:vrb(<vscale x 2 x s1>) = G_ICMP intpred(ult), %0(<vscale x 2 x s8>), %0
```
Comment (topperc) [inline llvm/test/CodeGen/RISCV/GlobalISel/instruction-select/rvv/icmp.mir:50]: Test other condition codes than just ult

**Review by topperc — APPROVED** [review]
LGTM

**Comment by michaelmaitland** [comment]
Closed by 05f673bcefb0912a38a67b0026cad3768b2f85d2 and 35a9393a3f775d4e1506965b9cfeedd45599f1a7

**Pull request closed without merging**